File: Algorithms/bubble_sort.c
#include <stdio.h>

void bubble_sort(int arr[], int N) {
	for (int i = 0; i < N; i++) {
		for (int j = 1; j < N; j++) {
			// Swap current element with previous element, if unordered
			if (arr[j-1] > arr[j]) {
				int temp = arr[j-1];
				arr[j-1] = arr[j];
				arr[j] = temp;
			}
		}
	}
}

int main() {
	int arr[] = {5,74,12,9,0,55,19,-5,8,4};
	// Calculate array length by dividing total size by size of each element
	int N = sizeof(arr) / sizeof(arr[0]);

	// Print elements of unsorted array
	printf("Unsorted array:");
	for (int i = 0; i < N; i++)
		printf(" %d", arr[i]);
	printf("\n");

	bubble_sort(arr, N);

	// Print elements of sorted array
	printf("Sorted array:  ");
	for (int i = 0; i < N; i++)
		printf(" %d", arr[i]);
	printf("\n");

	return 0;
}

File: Sorting/BubbleSort/bubble_sort.py
# Runtime Complexity: O(N^2)
# Space Complexity: O(1)
def bubble_sort(arr):
	N = len(arr)
	for i in range(N):
		for j in range(1, N - i):
			# Swap current element with previous element, if unordered
			if arr[j-1] > arr[j]:
				temp = arr[j-1]
				arr[j-1] = arr[j]
				arr[j] = temp

def main():
	arr = [5,74,12,9,0,55,19,-5,8,4]
	print(f"Unsorted array: {arr}")
	bubble_sort(arr)
	print(f"Sorted array:   {arr}")

if __name__ == "__main__":
	main()
